api_controller/models.py

```diff
@@ -1,139 +1,201 @@
+            response = requests.get(api_url,
+                                    headers={'Authorization': 'Bearer ' + self.token}).json()
+            objs += response['{}s'.format(obj_class)]
+        if keep_order:
+            return sorted(objs, key=lambda x: obj_ids.index(x['id']))
+        return objs
+
     def fetch_stepik_step(self, pk):
         return self.fetch_stepik_object("step", pk)
 
     def fetch_stepik_lesson(self, pk):
         return self.fetch_stepik_object("lesson", pk)
 
-    def steps_text(self, ids):
+    # :returns: list of tuples(text, update_date)
+    def steps_text_with_dates(self, ids):
         texts = []
-        for id in ids:
-            step = self.fetch_stepik_object("step", id)
-            if step is None or 'text' in step['block']:
+        steps = self.fetch_stepik_objects("step", ids)
+        for step in steps:
+            if step is None or 'text' not in step['block']:
                 texts.append("")
             else:
-                texts.append(step['block']['text'])
+                texts.append((step['block']['text'], self.from_str_to_datetime(step['update_date'])))
         return texts
 
+    # :returns: created translation object or None if service_name is None or can't be parsed or
+    # if Stepik API returned 404 or have no permission
     def create_translation(self, obj_type, pk, service_name=None, lang=None):
-        if self.translation_services.filter(service_name=service_name.lower()).exists():
-            translation_service = self.translation_services.get(service_name=service_name.lower())
-        else:
+        translation_service = self.get_service(service_name)
+        if not translation_service:
             return None
 
-        if obj_type == "steps":
+        if obj_type is RequestedObject.STEP:
+
             translation = self.get_translation(obj_type, pk, service_name, lang)
             if translation is not None:
                 return translation
-            created = self.fetch_stepik_object(obj_type[:-1], pk)
+            created = self.fetch_stepik_object(obj_type.value, pk)
             if created is None:
                 return None
-            ts = None
-            translated_text = translation_service.create_step_translation(created['block']['text'], lang=lang)
-            tl = TranslatedLesson.objects.create(stepik_id=created['lesson'], service_name=service_name)
-            ts = TranslationStep.objects.create(stepik_id=pk, lang=lang, service_name=service_name,
-                                                text=translated_text, lesson=tl)
-
-            return ts
-        elif obj_type == "lessons":
+            created_step, lesson_keeper = None, None
+            lesson = self.fetch_stepik_object("lesson", created['lesson'])
+            translated_text = translation_service.create_text_translation(created['block']['text'], lang=lang)
```

Comment:
@KirillMakarov yeah, good catch, but yes. Above we check if step is None of not, otherwise step with video has empty string